programs/pkey/CMakeLists.txt

```diff
@@ -1,63 +1,71 @@
 add_executable(dh_client dh_client.c)
 target_link_libraries(dh_client mbedtls)
 
-add_executable(dh_genprime dh_genprime.c)
-target_link_libraries(dh_genprime mbedtls)
-
 add_executable(dh_server dh_server.c)
 target_link_libraries(dh_server mbedtls)
 
-add_executable(ecdh_curve25519 ecdh_curve25519.c)
-target_link_libraries(ecdh_curve25519 mbedtls)
+if(!ENABLE_PSA)
```

Comment:
I don't think we need to do this. Why are we disabling them?

Reply:
If these programs are already part of the crypto submodule, why include them? We are not including crypto submodule tests, so we can exclude crypto submodule programs as well.

Reply:
I think this is about transitioning to the Crypto module. As I see it, is goes in phases:

1. We add the submodule. There's duplication of some files. That's fine. Users get Mbed TLS as it was, and the option of the new stuff.
2. We remove the duplicates, and migrate stuff to the new module. We remove duplicated programs, test and library files.

We're working on 1. At the the moment if a program is duplicated, so what? At the moment if users want Mbed TLS they will expect the Mbed TLS programs to build. If they want PSA, they need to cd into the directory and build that directly. I think doing this before step 2 is confusing.